### PR TITLE
feat(toast): add ability to use toast inline

### PR DESCRIPTION
**Pull request by amandaejohnston**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Toast can only be presented using `toastController`.

Issue URL: N/A

## What is the new behavior?

Toast can now be used as an inline component, with support for both `isOpen` and `trigger`.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/react/src/components/IonToast.tsx [hunk 1]
```diff
@@ -20,15 +20,7 @@ export interface ToastOptions extends Omit<ToastOptionsCore, 'buttons'> {
...
 
-const toastController = {
-  create: (options: ToastOptions) => toastControllerCore.create(options as any),
-  dismiss: (data?: any, role?: string | undefined, id?: string | undefined) =>
-    toastControllerCore.dismiss(data, role, id),
-  getTop: () => toastControllerCore.getTop(),
-};
-
-export const IonToast = /*@__PURE__*/ createControllerComponent<ToastOptions, HTMLIonToastElement>(
+export const IonToast = /*@__PURE__*/ createInlineOverlayComponent<JSX.IonActionSheet, HTMLIonActionSheetElement>(
```
Comment (sean-perkins) [inline packages/react/src/components/IonToast.tsx:23]: The React build is likely failing from this copy & paste. Should reference the Toast component and HTML element type.

**Review by sean-perkins — COMMENTED** [review]
The Vue failing build likely needs to update the e2e test to use the `testInlineOverlay` function for the component test.

File: packages/react/src/components/IonToast.tsx (hunk 1 above)
Comment (amandaejohnston) [reply packages/react/src/components/IonToast.tsx:23]: Ah, good catch 👀

**Review by sean-perkins — APPROVED** [review]
Looks good!

**Review by liamdebeasi — APPROVED** [review]
Great job!

**Pull request merged**